src/common/config_parser.cpp

```diff
@@ -635,6 +634,8 @@ void ConfigParser::parseOptions(int argc, char** argv, bool doValidate) {
   if(get<bool>("version")) {
     std::cerr << PROJECT_VERSION_FULL << std::endl;
     exit(0);
+  } else {
+    config_["version"] = PROJECT_VERSION_FULL;
```

Comment:
Maybe add a comment. It took me a moment to understand what this does.

Reply:
Now, `config["version"]` is a version number and not related to the --version flag